File: Dao/pedidos-react-23-main/pedidos-react-23-main/src/services/pedidoService.js
import api from "./pedidosAPI";

class PedidoService {
  buscarTodos() {
    return api.get("/pedidos");
  }

  buscarPorId(id) {
    return api.get(`/pedidos/${id}`);
  }

  crear(data) {
    return api.post("/pedidos", data);
  }

  modificar(id, data) {
    return api.put(`/pedidos/${id}`, data);
  }

  eliminar(id) {
    return api.delete(`/pedidos/${id}`);
  }

  buscarPorClienteDNI(dni) {
    return api.get(`/pedidos?clienteDni=${dni}`);
  }
}

const service = new PedidoService();
export default service

File: Dao/pedidos-rest-24-main/pedidos-rest-24-main/src/main/java/es/uvigo/mei/pedidos/servicios/PedidoService.java
package es.uvigo.mei.pedidos.servicios;

import java.util.List;
import java.util.Optional;

import es.uvigo.mei.pedidos.entidades.Pedido;
import es.uvigo.mei.pedidos.entidades.Cliente;

public interface PedidoService {
	public Pedido crear(Pedido pedido);
	public Pedido modificar(Pedido pedido);
	public void eliminar(Pedido pedido);
	public Optional<Pedido> buscarPorId(Long id);
	public Pedido buscarPorIdConLineas(Long id);	
	public List<Pedido> buscarTodos();
	public List<Pedido> buscarPorCliente(Cliente cliente);

}
